src/main/java/org/osiam/addons/self_administration/Config.java

```diff
@@ -118,43 +131,114 @@
+            }
+            if (!FIELDS.contains(htmlField.getName())) {
+                LOGGER.warn("The configured attribute '{}' is not available and will be ignored.", htmlField.getName());
+            }
+            collectedFields.add(htmlField);
         }
 
-        if (!collectedFields.contains("email")) {
-            collectedFields.add("email");
+        if (!containsField(collectedFields, "email")) {
+            htmlField = new HtmlField();
+            htmlField.setName("email");
+            htmlField.setRequired(true);
+            collectedFields.add(htmlField);
         }
 
-        if (!collectedFields.contains("password")) {
-            collectedFields.add("password");
+        if (!containsField(collectedFields, "password")) {
+            htmlField = new HtmlField();
+            htmlField.setName("password");
+            htmlField.setRequired(true);
+            collectedFields.add(htmlField);
         }
 
-        if (!collectedFields.contains("confirmPassword")) {
+        final HtmlField confirmPasswordField = getField(collectedFields, "confirmPassword");
+        if (confirmPasswordField != null) {
+            confirmPasswordField.setRequired(true);
+        } else {
             confirmPasswordRequired = false;
         }
 
         String fieldUserName = "userName";
-        if (!usernameEqualsEmail && !collectedFields.contains(fieldUserName)) {
-            collectedFields.add(fieldUserName);
-        } else if (usernameEqualsEmail && collectedFields.contains(fieldUserName)) {
-            collectedFields.remove(fieldUserName);
+        if (!usernameEqualsEmail && !containsField(collectedFields, fieldUserName)) {
+            htmlField = new HtmlField();
+            htmlField.setName(fieldUserName);
+            htmlField.setRequired(false);
+            collectedFields.add(htmlField);
+        } else if (usernameEqualsEmail && containsField(collectedFields, fieldUserName)) {
+            removeField(collectedFields, fieldUserName);
         }
 
+        // Example extensions['urn:client:extension'].fields['age']:true:number
         for (String field : extensions) {
-            collectedFields.add(field.trim());
+            htmlField = new HtmlField();
+            String trimmedField = field.trim();
+            if (trimmedField.contains(":")) {
+                int lastIndexOf = trimmedField.lastIndexOf(":");
+                if (lastIndexOf == -1) {
+                    throw new ConfigurationException("The extension configuration must be in the form field:required " +
+                            "e.g. 'extensions['urn:client:extension'].fields['age']:true' or just the field e.g. " +
+                            "'extensions['urn:client:extension'].fields['age']', but it is: " + trimmedField);
+                }
+
+                final String type = trimmedField.substring(lastIndexOf + 1, trimmedField.length());
+                final boolean required;
+                // urn:required
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/t-pick/addon-self-administration/file/2113712806/issues/source?bid=3145222&fileBranchId=3145225#l215)